geps/gep-3539/index.md

```diff
@@ -0,0 +1,200 @@
+cases. More complex traffic splitting and advanced load balancing cases can still use Route attachments via 
+allowedRoutes. 
+
+### DNS
+
+ClusterIP Gateways in the cluster need to have consistent DNS names assigned to allow ClusterIP lookup by 
+name rather than IP address. DNS A and/or AAAA record creation needs to happen when Kubernetes publishes 
+information about Gateways, in a manner similar to ClusterIP Service creation behavior. DNS nameservers 
+in pods’ /etc/resolv.conf need to be programmed accordingly by kubelet.
+
+```
+<name of gateway>.<gateway-namespace>.gw.cluster.local
+```
+
+This results in the following search option entries in Pods’ /etc/resolv.conf:
+```
+search <ns>.gw.cluster.local  gw.cluster.local  cluster.local
+```
+
+### Cross-namespace References
+
+Gateway API allows for Routes in different namespaces to attach to the Gateway. 
+
+When modeling ClusterIP service networking, the simplest recommendation might be to keep Gateway and Routes 
+within the same namespace. While cross namespace routing would work and allow for evolved functionality, 
+it may make supporting certain cases tricky. One specific example for this case is the pod DNS resolution 
+support of the following format 
+
+```
+pod-ipv4-address.gateway-name.my-namespace.gw.cluster-domain.example
+```
+
+If Gateway and Routes (and hence the backing pods) are in different namespaces, there arises ambiguity in 
+whether and how to support this pod DNS resolution format.
+
+## LoadBalancer and NodePort Services
+
+Extending the concept further to LoadBalancer and NodePort type services follows a similar pattern. The idea 
+is to have a GatewayClass corresponding to each type of service networking behavior that needs to be modeled 
+and supported.
+
+![image displaying gatewayclasses to represent different service types](images/gatewayclasses-lb-np.png "image displaying gatewayclasses to represent different service types")
+
+Note that Gateway API allows flexibility and clear separation of concerns so that one would not need to 
+configure cluster-ip and node-port when configuring a load-balancer.
+
+But for completeness, the case shown below demonstrates how load balancer functionality analogous to 
+LoadBalancer Service API can be achieved using Gateway API.
+
+![image displaying LB Service API analogous GW API objects](images/LB-NP-Clusterip.png "image displaying LB Service API analogous GW API objects")
+
+## Additional Service API Features
+
+Services natively provide additional features as listed below (not an exhaustive list). Gateway API can be 
+extended to provide some of these features natively, while others may be left up to the specifics of 
+implementations. 
+
+| Feature | ServiceAPI options | Gateway API possibilities |
+|---|---|---|
+| sessionAffinity | ClientIP <br /> NoAffinity | Route level
+| allocateLoadBalancerNodePorts | True <br /> False | Not supported for ClusterIP Gateway <br /> Supported for LoadBalancer Gateway |
```

Comment:
we could say N/A for this approach, since you can create LB type without NodePort - sort of simplification.

Reply:
That might be clearer until further discussion on each of these.